app/v5/bfLosses/list/model/response/ListBFLossesItem.scala

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright 2024 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package v5.bfLosses.list.model.response
+
+import v5.bfLosses.list.model.TypeOfLoss
+
+trait ListBFLossesItem {
```

Comment:
So I've removed the trait but I couldn't see the bit where the service layer is referring to an "isEmpty". The closest for this is an emptiness check in the second clause that we check for with the library function isEmpty. Am I missing something here?

Reply:
But now `ListBFLossesResponse` refers to `v5.bfLosses.list.def1.model.response.ListBFLossesItem` (in def1). The choice you have is:

1. Remove `losses` from the  `ListBFLossesResponse` trait completely. In this case [the service](https://github.com/hmrc/individual-losses-api/blob/eb9067cb57c4f46b8d977978ee0543266beb9938/app/v5/bfLosses/list/ListBFLossesService.scala#L38) will need to determine whether it is empty. So you could have an abstract `isEmpty` method on `ListBFLossesResponse` to do this.
2. Re-instate `ListBFLossesItem` (not in def1) and a `Def1_ListBFLossesItem`. But, crucially, `ListBFLossesItem` would be an empty trait so that we don't constrain the fields for def2, def3 etc.